src/shared/loop-util.c

```diff
@@ -104,26 +111,40 @@ static int device_has_block_children(sd_device *d) {
                 return r;
 
         FOREACH_DEVICE(e, q) {
-                const char *ss, *sn;
+                const char *ss, *sn, *dt;
+
+                r = sd_device_get_sysname(q, &sn);
+                if (r < 0) {
+                        log_debug_errno(r, "Failed to get sysname of child, ignoring: %m");
+                        continue;
+                }
 
                 r = sd_device_get_subsystem(q, &ss);
-                if (r < 0)
+                if (r < 0) {
+                        log_debug_errno(r, "Failed to get subsystem of child '%s', ignoring: %m", sn);
                         continue;
+                }
 
-                if (!streq(ss, "block"))
+                if (!streq(ss, "block")) {
+                        log_debug("Skipping child '%s' that is not a block device (subsystem=%s).", sn, ss);
                         continue;
+                }
 
-                r = sd_device_get_sysname(q, &sn);
-                if (r < 0)
+                r = sd_device_get_devtype(q, &dt);
+                if (r < 0) {
+                        log_debug_errno(r, "Failed to get devtype of child '%s', ignoring: %m", sn);
                         continue;
+                }
 
-                if (streq(sn, main_sn))
```

Comment:
So, this spurious check is dropped. So, sn and main_sn are not used any more, and can be dropped.

Note, sn is used for logging, but you can use log_device_debug_errno().